igvm/hypervisor_preferences.py

```diff
@@ -130,6 +130,36 @@ def __call__(self, vm, hv):
         return value is not None and value > self.limit
 
 
+class HypervisorCpuUsageLimit(object):
+    """Check for CPU usage of the hypervisor incl. the predicted CPU usage
+    of the VM to be migrated.
+
+    Dismiss any hypervisor, which would be above its threshold.
+    """
+    def __init__(self, hardware_model, hv_cpu_thresholds):
+        self.hardware_model = hardware_model
+        self.hv_cpu_thresholds = hv_cpu_thresholds
+
+    def __repr__(self):
+        args = repr(self.hardware_model) + ', ' + repr(self.hv_cpu_thresholds)
+
+        return '{}({})'.format(type(self).__name__, args)
+
+    def __call__(self, vm, hv):
+        # New VM has no hypervisor attribute yet.
+        if not vm.hypervisor:
+            return False
+
+        hv_model = hv.dataset_obj[self.hardware_model]
+        hv_cpu_threshold = self.hv_cpu_thresholds[hv_model]
+        hv_cpu_util_overall = hv.hv_cpu_util_overall(vm)
+
+        return (
+                hv_cpu_util_overall is not None
```

Comment:
What happens if hv_cpu_util_overall is None ? The return value of the method would be None because none of the return statements terminated the method. What happens in such as case should we maybe return True or False instead ?

Reply:
If `hv_cpu_util_overall` is None, we're automatically returning `False` in every condition
```
>>> a = (None is not None and 78.5 > 70.0)
>>> a
False
>>> a = (62.5 is not None and 62.5 > 70.0)
>>> a
False
>>> a = (78.5 is not None and 78.5 > 70.0)
>>> a
True
```

Only `True` would put the HV to the end of the list.

Reply:
Ah true I was confused.